Stripe/STPBINRange.m

```diff
@@ -30,22 +30,34 @@ @implementation STPBINRange
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         NSArray *ranges = @[
-                            // Catch-all values
                             @[@"", @"", @16, @(STPCardBrandUnknown)],
+
                             @[@"34", @"34", @15, @(STPCardBrandAmex)],
                             @[@"37", @"37", @15, @(STPCardBrandAmex)],
-                            @[@"30", @"30", @14, @(STPCardBrandDinersClub)],
-                            @[@"36", @"36", @14, @(STPCardBrandDinersClub)],
-                            @[@"38", @"39", @14, @(STPCardBrandDinersClub)],
-                            @[@"60", @"60", @16, @(STPCardBrandDiscover)],
-                            @[@"62", @"62", @16, @(STPCardBrandDiscover)],
-                            @[@"64", @"65", @16, @(STPCardBrandDiscover)],
+
+                            @[@"30000000", @"30599999", @14, @(STPCardBrandDinersClub)],
+                            @[@"30950000", @"30959999", @14, @(STPCardBrandDinersClub)],
+                            @[@"36000000", @"36999999", @14, @(STPCardBrandDinersClub)],
+                            @[@"38000000", @"39999999", @14, @(STPCardBrandDinersClub)],
+
+                            @[@"60110000", @"60110999", @16, @(STPCardBrandDiscover)],
+                            @[@"61120000", @"60114999", @16, @(STPCardBrandDiscover)],
+                            @[@"60117400", @"60117499", @16, @(STPCardBrandDiscover)],
+                            @[@"60117700", @"60117999", @16, @(STPCardBrandDiscover)],
+                            @[@"60118600", @"60119999", @16, @(STPCardBrandDiscover)],
+                            @[@"64400000", @"65999999", @16, @(STPCardBrandDiscover)],
+
                             @[@"35", @"35", @16, @(STPCardBrandJCB)],
+
                             @[@"50", @"59", @16, @(STPCardBrandMasterCard)],
                             @[@"22", @"27", @16, @(STPCardBrandMasterCard)],
                             @[@"67", @"67", @16, @(STPCardBrandMasterCard)], // Maestro
+
+                            @[@"62212600", @"62292599", @16, @(STPCardBrandUnionPay)],
```

Comment:
I think we should just use [@"62", @"62", @16, @(STPCardBrandUnionPay)] to be more permissive